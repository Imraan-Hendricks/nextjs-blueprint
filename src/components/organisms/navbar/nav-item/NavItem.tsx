import Link from 'next/link';
import { clsx } from 'clsx';

interface NavItemProps {
  active?: boolean;
  children: string;
  href: string;
}

export function NavItem({ active, children, href }: NavItemProps) {
  return (
    <li>
      <Link
        href={href}
        className={clsx(
          'w-max cursor-pointer rounded px-4 py-2 transition-colors',
          'hover:bg-indigo-500 hover:text-white active:bg-indigo-400',
          active ? 'text-indigo-600' : 'text-gray-500',
        )}>
        {children}
      </Link>
    </li>
  );
}
