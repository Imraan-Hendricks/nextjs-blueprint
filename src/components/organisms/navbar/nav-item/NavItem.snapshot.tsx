import { cleanup, render } from '@testing-library/react';
import { NavItem } from './NavItem';

describe('NavItem Snapshot Tests', () => {
  beforeAll(() => {
    cleanup();
  });

  afterEach(() => {
    cleanup();
  });

  test('renders correctly with active prop', () => {
    const { container } = render(
      <NavItem href='/some-link' active>
        My Link
      </NavItem>,
    );
    expect(container).toMatchSnapshot();
  });

  test('renders correctly without active prop', () => {
    const { container } = render(<NavItem href='/some-link'>My Link</NavItem>);
    expect(container).toMatchSnapshot();
  });
});
