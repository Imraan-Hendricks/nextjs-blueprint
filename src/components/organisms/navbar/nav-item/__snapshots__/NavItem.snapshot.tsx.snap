// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem Snapshot Tests renders correctly with active prop 1`] = `
<div>
  <li>
    <a
      class="w-max cursor-pointer rounded px-4 py-2 transition-colors hover:bg-indigo-500 hover:text-white active:bg-indigo-400 text-indigo-600"
      href="/some-link"
    >
      My Link
    </a>
  </li>
</div>
`;

exports[`NavItem Snapshot Tests renders correctly without active prop 1`] = `
<div>
  <li>
    <a
      class="w-max cursor-pointer rounded px-4 py-2 transition-colors hover:bg-indigo-500 hover:text-white active:bg-indigo-400 text-gray-500"
      href="/some-link"
    >
      My Link
    </a>
  </li>
</div>
`;
